download-utils/src/lib.rs

```diff
@@ -108,34 +106,13 @@ pub fn download_file(url: &str, destination_file: &Path) -> Result<(), String> {
     Ok(())
 }
 
-fn extract_archive(archive_filename: &Path, destination_dir: &Path) -> Result<(), String> {
-    info!("Extracting {:?}...", archive_filename);
-    let extract_start = Instant::now();
-
-    fs::create_dir_all(destination_dir).map_err(|err| err.to_string())?;
-    let tar_bz2 = File::open(&archive_filename)
-        .map_err(|err| format!("Unable to open {:?}: {:?}", archive_filename, err))?;
-    let tar = BzDecoder::new(std::io::BufReader::new(tar_bz2));
-    let mut archive = tar::Archive::new(tar);
-    archive
-        .unpack(destination_dir)
-        .map_err(|err| format!("Unable to unpack {:?}: {:?}", archive_filename, err))?;
-    info!(
-        "Extracted {:?} in {:?}",
-        archive_filename,
-        Instant::now().duration_since(extract_start)
-    );
-    Ok(())
-}
-
-pub fn download_genesis(
+pub fn download_genesis_if_missing(
     rpc_addr: &SocketAddr,
     ledger_path: &Path,
-    expected_genesis_hash: Option<Hash>,
-) -> Result<Hash, String> {
+) -> Result<Option<(std::path::PathBuf, std::path::PathBuf)>, String> {
```

Comment:
nit: `use std::path`?

Reply:
The return type feels too complex here.  The caller doesn't differentiate between `Err(_)` and `Ok(None)`, so perhaps this can be simplified to `Result<PathBuf,String>`, where `Ok(_)` returns the `tmp_genesis_package`

Reply:
nice!